### PR TITLE
Make the release checklist shorter and hide some details

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

There is a lot of detail in the release checklist, we can hide or remove some of it.

[Rendered](https://github.com/ZcashFoundation/zebra/blob/short-release-checklist/.github/PULL_REQUEST_TEMPLATE/release-checklist.md)

## Solution

- Remove repeated information
- Hide some detailed examples and explanations
- Fix some markdown formatting issues

## Review

Anyone can review this PR.

### Reviewer Checklist

  - [ ] New checklist makes sense

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -62,35 +86,23 @@ You can use `fastmod` to interactively find and replace versions.
 
 For example, you can do something like:
 ```
-fastmod --extensions rs,toml,md --fixed-strings '1.0.0-alpha.12' '1.0.0-alpha.13'
-fastmod --extensions rs,toml,md --fixed-strings '0.2.9' '0.2.10' tower-batch
-fastmod --extensions rs,toml,md --fixed-strings '0.2.8' '0.2.9' tower-fallback
+fastmod --extensions rs,toml,md --fixed-strings '1.0.0-beta.11' '1.0.0-beta.12'
+fastmod --extensions rs,toml,md --fixed-strings '0.2.26' '0.2.27' tower-batch tower-fallback
```
Comment (oxarbitrage) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md:90]: ` tower-batch` and `tower-fallback` has different versions.

https://github.com/ZcashFoundation/zebra/blob/main/tower-batch/Cargo.toml#L3
https://github.com/ZcashFoundation/zebra/blob/main/tower-fallback/Cargo.toml#L3

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -11,8 +11,19 @@ assignees: ''
 
 ### Which Crates to Increment
 
-To check if any of the top-level crates need version increments, go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra and use the last modified dates of each crate. Alternatively you can use the github compare tool and check the `main` branch against the last tag ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.15...main)). `git diff --stat <previous-release-tag> origin/main` is also useful to see what's changed.
-
+To check if any of the top-level crates need version increments:
+1. go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra
```
Comment (oxarbitrage) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: There are list items that start with capital letters and others do not. Do you think we should start all list items of the template with a capital letter (where possible) ?

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -22,6 +33,8 @@ To check if any of the top-level crates need version increments, go to the zebra
 Zebra follows [semantic versioning](https://semver.org).
 
 Semantic versions look like: `MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]`
```
Comment (oxarbitrage) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: It seems this format is not correct (or not the one we are using). Can we adjust it while we are here?

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 4]
```diff
@@ -62,35 +86,23 @@ You can use `fastmod` to interactively find and replace versions.
...
-### Reviewing Version Bumps
-
-Check for missed changes by going to:
-`https://github.com/ZcashFoundation/zebra/tree/<commit-hash>/`
-Where `<commit-hash>` is the hash of the last commit in the version bump PR.
-
-If any Zebra or Tower crates have commit messages that are **not** a version bump, we have missed an update.
-
-Also check for crates that depend on crates that have changed. They should get a version bump as well.
+Ignore any `fastmod` updates to previous release notes in `CHANGELOG.md`.
```
Comment (oxarbitrage) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: It is not clear what this sentence means to me.

**Review by oxarbitrage — COMMENTED** [review]
It looks good, the `<detail>` thing is nice, i didn't knew it was possible. I left a few comments but the only important one is the fastmod command one.

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (teor2345) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md:90]: I gave them the same version in the latest release, to simplify the release process.

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md (hunk 2 above)
Comment (teor2345) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: Thanks, fixed in c8cb7a300672c0f1899216e18dac49d297831c5e

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md (hunk 4 above)
Comment (teor2345) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: Fixed in 181cd649e6e83893fa6f9fb0be8fd94ecefd3056

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md (hunk 3 above)
Comment (teor2345) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: I think the format is correct, but the highlighting is confusing:

```suggestion
Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]
```

Comment (teor2345) [reply .github/PULL_REQUEST_TEMPLATE/release-checklist.md]: There is a more detailed explanation inside the `<details>` block, which shows examples of each of the components.

**Review by oxarbitrage — APPROVED** [review]
thanks for the changes.

**Pull request merged**